pkg/skaffold/watch/changes.go

```diff
@@ -17,60 +17,96 @@ limitations under the License.
 package watch
 
 import (
+	"fmt"
 	"os"
+	"sort"
+	"strings"
 	"time"
 
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 )
 
-type fileMap struct {
-	count        int
-	lastModified time.Time
-}
+type fileMap map[string]time.Time
 
+// TODO(mrick): cached tree extension ala git
 func stat(deps func() ([]string, error)) (fileMap, error) {
+	state := fileMap{}
 	paths, err := deps()
 	if err != nil {
-		return fileMap{}, errors.Wrap(err, "listing files")
+		return state, errors.Wrap(err, "listing files")
 	}
-
-	last, err := lastModified(paths)
-	if err != nil {
-		return fileMap{}, err
+	for _, path := range paths {
+		stat, err := os.Stat(path)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to stat file %s", path)
+		}
+		state[path] = stat.ModTime()
 	}
 
-	return fileMap{
-		count:        len(paths),
-		lastModified: last,
-	}, nil
+	return state, nil
 }
 
-func lastModified(paths []string) (time.Time, error) {
-	var last time.Time
+type Events struct {
+	Added    []string
+	Modified []string
+	Deleted  []string
+}
 
-	for _, path := range paths {
-		stat, err := os.Stat(path)
-		if err != nil {
-			if os.IsNotExist(err) {
```

Comment:
This test should be kept. And we might want to had a unit test to protect against breaking it.
Watch shouldn't fail on a broken link or if a file disappear just after the artifact says it relies on this file.

Reply:
Good catch. Added it back and added some explicit test coverage of those cases.